cpp/oneapi/dal/algo/knn/infer_types.cpp

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/algo/knn/infer_types.hpp"
+
+namespace oneapi::dal::knn {
+
+class detail::infer_input_impl : public base {
+  public:
+    infer_input_impl(const table& query, model* m)
+        : query_(query), model_(m) {}
+
+    table query_;
+    model* model_;
+};
+
+class detail::infer_result_impl : public base {
+  public:
+    table classes_;
```

Comment:
```suggestion
    table labels_;
```
It seems to me more understandable? And described in the specification.